packages/frontend/src/components/MetricQueryData/MetricQueryDataProvider.tsx

```diff
@@ -69,11 +69,15 @@ export const getDataFromChartClick = (
         (item) => !isDimension(item),
     );
 
-    const selectedField =
-        selectedMetricsAndTableCalculations.length > 0
-            ? selectedMetricsAndTableCalculations[0]
-            : selectedFields[0];
-    const selectedValue = e.data[getItemId(selectedField)];
+    let selectedField: Field | TableCalculation | undefined = undefined;
+    if (selectedMetricsAndTableCalculations.length > 0) {
+        selectedField = selectedMetricsAndTableCalculations[0];
+    } else if (selectedFields.length > 0) {
+        selectedField = selectedFields[0];
+    }
+    const selectedValue = selectedField
+        ? e.data[getItemId(selectedField)]
+        : undefined;
     const fieldValues: Record<string, ResultValue> = Object.entries(
         e.data as Record<string, any>,
```

Comment:
the check here <string, any> is making the whole thing unreliable including the return value. im wondering if returning `value: { raw: selectedValue, formatted: selectedValue },` with undefined is going to break something elsewhere. it's probably worth it to replace  `Record<string, any> `with  `Record<string, ResultValue>`
and make sure returned type satisfies places where `getDataFromChartClick` is being called atm